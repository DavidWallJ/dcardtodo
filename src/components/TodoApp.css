.container {
	height: 100%;
	display: flex;
	justify-content: center;
}

.todoAppContainer {
	background-color: #f9f9f9;
	margin-top: 10%;
	padding: 10px;
	border-radius: 26px;
}

.todoAppContainer h1 {
	color: #4d4d4d;
	padding-bottom: 14px;
	font-family: 'Lato', sans-serif;
}
