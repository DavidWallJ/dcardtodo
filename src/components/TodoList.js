import React, { Component } from 'react';

import TodoItem from './TodoItem';

class TodoList extends Component {
	renderTodos = () => {
		const { todos } = this.props;
		return todos.map(todo => {
			return <TodoItem key={todo.id} {...todo} />;
		});
	};
	render() {
		return (
			<div>
				{this.renderTodos()}
			</div>
		);
	}
}

export default TodoList;
