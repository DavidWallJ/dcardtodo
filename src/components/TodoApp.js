import React, { Component } from 'react';
import { Panel } from 'react-bootstrap';
import uuid from 'node-uuid';
import moment from 'moment';

import TodoAPI from '../api/TodoAPI';
import TodoList from './TodoList';
import AddTodo from './AddTodo';
import TodoSearch from './TodoSearch';

class TodoApp extends Component {
	constructor(props) {
		super(props);
		this.state = {
			showCompleted: false,
			searchInput: '',
			todos: TodoAPI.getTodos()
		};
	}

	componentDidUpdate = () => {
		TodoAPI.setTodos(this.state.todos);
	};

	handleAddTodo = text => {
		this.setState({
			todos: [
				...this.state.todos,
				{
					id: uuid(),
					text,
					completed: false,
					createdAt: moment().unix(),
					completedAt: undefined
				}
			]
		});
	};

	handleSearch = searchInput => {
		console.log('handleSearch: ', searchInput);
		this.setState({
			searchInput: searchInput.toLowerCase()
		});
	};

	handleCheckbox = showCompleted => {
		console.log('handleCheckbox: ', showCompleted);
		this.setState({
			showCompleted: showCompleted
		});
	};

	handleToggle = id => {
		const updatedTodos = this.state.todos.map(todo => {
			if (todo.id === id) {
				todo.completed = !todo.completed;
				// set completedAt date if todo.completed is true
				todo.completedAt = todo.completed ? moment().unix() : undefined;
			}
			return todo;
		});

		this.setState({ todos: updatedTodos });
	};

	render() {
		const { todos, showCompleted, searchInput } = this.state;
		const filteredTodos = TodoAPI.filterTodos(
			todos,
			showCompleted,
			searchInput
		);
		return (
			<Panel header="Dcard Todo Application">
				<TodoSearch
					onSearch={this.handleSearch}
					onCheckbox={this.handleCheckbox}
					checked={this.state.showCompleted}
				/>
				<TodoList todos={filteredTodos} onToggle={this.handleToggle} />
				<AddTodo onAddTodo={this.handleAddTodo} />
			</Panel>
		);
	}
}

export default TodoApp;
