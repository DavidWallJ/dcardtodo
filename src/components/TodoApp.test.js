import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import TodoApp from './TodoApp';
import TodoList from './TodoList';

describe(TodoApp, () => {
	const component = shallow(<TodoApp />);

	it('should exist', () => {
		const component = renderer.create(<TodoApp />);
		const tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
