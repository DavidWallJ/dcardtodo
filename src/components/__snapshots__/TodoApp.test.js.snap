// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`class TodoApp extends _react.Component {
	constructor(props) {
		super(props);
		this.state = {
			todos: [
			{
				id: 1,
				text: 'Talk the talk' },

			{
				id: 2,
				text: 'Walk the walk' },

			{
				id: 3,
				text: 'Water the cat' },

			{
				id: 4,
				text: 'Buy water for cat' }] };



	}
	render() {
		var { todos } = this.state;
		return _react2.default.createElement(_TodoList2.default, { todos: todos, __source: { fileName: _jsxFileName, lineNumber: 30 }, __self: this });
	}} should exist 1`] = `
<div>
  <div>
    <p>
      1
      . 
      Talk the talk
    </p>
  </div>
  <div>
    <p>
      2
      . 
      Walk the walk
    </p>
  </div>
  <div>
    <p>
      3
      . 
      Water the cat
    </p>
  </div>
  <div>
    <p>
      4
      . 
      Buy water for cat
    </p>
  </div>
</div>
`;
